src/System.Collections/tests/Generic/HashSet/HashSet_ConstructorTests.cs

```diff
@@ -328,6 +328,108 @@ public static void Ctor_IEnumerable_HashSet_Multiple_Neg()
             driver.VerifyHashSet_NegativeTests();
         }
 
+        [Fact]
+        public static void Ctor_IEnumerable_HashSet_Fresh_HashSet()
+        {
+            HashSet<int> source = new HashSet<int>();
+            HashSet<int> hashSet = new HashSet<int>(source);
+
+            HashSetTestSupport<int> driver = new HashSetTestSupport<int>(hashSet, s_intGenerator, new int[0]);
+            driver.VerifyHashSetTests();
+        }
+
+        private static bool IsSquare(int value)
+        {
+            if (value == 0)
+                return true;
+            switch(value & 0x3F)
+            {
+                case 0:
+                case 1:
+                case 4:
+                case 9:
+                case 16:
+                case 17:
+                case 25:
+                case 33:
+                case 36:
+                case 41:
+                case 49:
+                case 57:
```

Comment:
It's known mathematically that any base that is a power of two will have a set number final digits that all square numbers end with (e.g. in hex, they all end with 0, 1, 4 or 9